### PR TITLE
Add getPromptAsync method to prompt selectors, deprecate existing sync methods in favor of async methods

### PR DESCRIPTION
**Pull request by jacoblee93**
Will allow us to apply partial variables to prompts selected with a `PromptSelector`.

### CODE REVIEW

**Comment by jacoblee93** [comment]
Will allow us to better support use cases like: #1176

File: docs/docs/modules/chains/prompt_selectors/index.mdx [hunk 1]
```diff
@@ -34,11 +34,11 @@ This will return `DEFAULT_QA_PROMPT` if the model is not a chat model, and `CHAT
 The example below shows how to use a prompt selector when loading a chain:
 
 ```typescript
-const loadQAStuffChain = (
+const initializeQAStuffChain = async (
   llm: BaseLanguageModel,
   params: StuffQAChainParams = {}
 ) => {
-  const { prompt = QA_PROMPT_SELECTOR.getPrompt(llm) } = params;
+  const { prompt = await QA_PROMPT_SELECTOR.getPromptAsync(llm) } = params;
```
Comment (nfcampos) [inline docs/docs/modules/chains/prompt_selectors/index.mdx:41]: I don't think we need to make this change now, we can do this in a separate PR
